lua/wire/wirenet.lua

```diff
@@ -0,0 +1,198 @@
+--[[
+	Consolidated networking library for Wiremod.
+	Trivial Networking - A wrapper for sending very small net messages that don't need a whole networkstring.
+]]
+
+local Net = WireLib.Net or {}
+WireLib.Net = Net
+
+-- Trivial Networking
+Net.Trivial = {}
+local SIZE = 10 -- 1023 names
+
+local registered_handlers = {}
+
+local update_handlers
+
+local function msg_handler(len, ply)
+	local handler = registered_handlers[net.ReadUInt(SIZE)]
+	if handler then
+		handler(len, ply)
+	else
+		error("WireLib.Net.Trivial tried to receive message that is not registered.")
+	end
+end
+
+net.Receive("wirelib_net_message", msg_handler)
+
+if SERVER then
+	util.AddNetworkString("wirelib_net_message")
+
+	local handler_names = {}
+	local handler_queue = {}
+	local queue_handler_firstidx, queue_handler_lastidx
+	local function queue_handler_flush(ply, tbl, first, last)
+		tbl = tbl or handler_queue
+		first = first or queue_handler_firstidx
+		last = last or queue_handler_lastidx
+
+		if first and first <= last then
+			net.Start("wirelib_net_message")
+				local data = table.concat(tbl, "\0") .. "\0"
```

Comment:
I don't think using null to separate messages is a good idea. Better to write the number of elements, write the length of each element, then the concat'd elements.

Reply:
What benefit does that give?

Reply:
If anything gets added to the table that contains a null character, it will corrupt the output